### PR TITLE
feat: #1411 clean up the webpack config for the project

### PR DESCRIPTION
**Pull request by tanphamhaiduong**
# Pull request checklist

## Does this close any currently open issues?

fixes: #1411 

**Please check if your PR fulfills the following requirements:**

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #1411

## What is the new behavior?

- Refactoring webpack config for centralize

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: scripts/webpack/webpack.config.dev.js [hunk 1]
```diff
@@ -118,6 +118,70 @@ const webpackConfig = {
...
+            loader: MiniCssExtractPlugin.loader,
+            options: {
+              hmr: process.env.NODE_ENV !== 'production',
+            },
+          },
+          'css-loader',
+        ],
+      },
+      {
+        test: /\.(sass|scss)$/,
```
Comment (phmngocnghia) [inline scripts/webpack/webpack.config.dev.js:135]: repeat css loader injected in dev and prod code, consider spliting them into that can be reused between dev and prod config

File: scripts/webpack/webpack.config.prod.js [hunk 2]
```diff
@@ -163,22 +101,81 @@ const webpackConfig = {
...
+            loader: 'css-loader',
             options: {
-              name: '[name].[ext]',
-              outputPath: '/assets',
+              importLoaders: 1,
             },
           },
         ],
       },
+      {
```
Comment (phmngocnghia) [inline scripts/webpack/webpack.config.prod.js:118]: See my comment about css loader above

File: package.json [hunk 3]
```diff
@@ -167,9 +167,6 @@
     "node-sass": "^4.12.0",
     "prettier": "^1.19.1",
     "prettier-plugin-packagejson": "^2.0.1",
-    "purgecss": "^1.3.0",
```
Comment (willmcvay) [inline package.json:170]: 🎉

File: scripts/webpack/webpack.config.dev.js (hunk 1 above)
Comment (tanphamhaiduong) [reply scripts/webpack/webpack.config.dev.js:135]: I think the reason why I separate prod and dev to avoid conflict for maintain in the future

**Pull request merged**